docs/core/StrategyManager.md

```diff
@@ -2,23 +2,32 @@
+| [`StrategyBaseTVLLimits.sol`](../../src/contracts/strategies/StrategyBaseTVLLimits.sol) | 3 instances (for cbETH, rETH, stETH) | Transparent proxy |
 
-<!-- Technical details on the LST subsystem as it functions during M2. Includes:
-* StrategyManager
-* Strategies (cbETH, rETH, stETH)
-* LST restaking
-* Stake / withdrawal flows -->
+The primary function of the `StrategyManager` is to handle accounting for individual Stakers as they deposit and withdraw LSTs from their corresponding strategies. It is responsible for (i) allowing Stakers to deposit/withdraw LSTs into the corresponding strategy, (ii) managing a queue of Staker withdrawals with an associated withdrawal delay, and (iii) keeping the `DelegationManager` updated as Stakers' shares change during these two operations.
 
-The primary function of the `StrategyManager` is to handle accounting for individual Stakers as they deposit and withdraw LSTs from their corresponding strategies. It is responsible for (i) allowing Stakers to deposit LST tokens into the corresponding strategy, (ii) managing a queue of Staker withdrawals with an associated withdrawal delay, and (iii) keeping the `DelegationManager` updated as Stakers' shares change during these two operations.
+As of M2, three LSTs are supported and each has its own instance of `StrategyBaseTVLLimits`: cbETH, rETH, and stETH. Each `StrategyBaseTVLLimits` has two main functions (`deposit` and `withdraw`), both of which can only be called by the `StrategyManager`. These `StrategyBaseTVLLimits` contracts are fairly simple deposit/withdraw contracts that hold tokens deposited by Stakers. Because these strategies are essentially extensions of the `StrategyManager`, their functions are documented in this file (see below).
 
-As of M2, three LSTs are supported and each has its own instance of `StrategyBaseTVLLimits`: cbETH, rETH, and stETH. Each `StrategyBaseTVLLimits` has two main functions (`deposit` and `withdraw`), both of which can only be called by the `StrategyManager`.
+*Important state variables*:
+* `mapping(address => mapping(IStrategy => uint256)) public stakerStrategyShares`: Tracks the current balance a Staker holds in a given strategy. Updated on deposit/withdraw.
+* `mapping(address => IStrategy[]) public stakerStrategyList`: Maintains a list of the strategies a Staker holds a nonzero number of shares in.
+    * Updated as needed when Stakers deposit and withdraw: if a Staker has a zero balance in a Strategy, it is removed from the list. Likewise, if a Staker deposits into a Strategy and did not previously have a balance, it is added to the list.
+* `mapping(bytes32 => bool) public withdrawalRootPending`: `QueuedWithdrawals` are hashed and set to `true` in this mapping when a withdrawal is initiated. The hash is set to false again when the withdrawal is completed. A per-staker nonce ensures that the same withdrawal cannot go through the queue twice.
```

Comment:
the nonce here is also used so people can, e.g., queue multiple otherwise identical withdrawals (e.g. 2 withdrawals of exactly 5 stETH)